packages/flutter/lib/src/material/snack_bar.dart

```diff
@@ -92,13 +92,19 @@ class SnackBarAction extends StatefulWidget {
     this.disabledTextColor,
     required this.label,
     required this.onPressed,
+    this.backgroundColor,
   }) : assert(label != null),
        assert(onPressed != null);
 
   /// The button label color. If not provided, defaults to
   /// [SnackBarThemeData.actionTextColor].
   final Color? textColor;
 
+  /// The button background fill color.
+  /// The state contains [MaterialState.disabled] after the [SnackBarAction]
+  /// is dismissed and can be used to show a different color after being dismissed.
+  final MaterialStateColor? backgroundColor;
```

Comment:
We support parameters with type `MaterialStateColor` in cases where there's an existing `Color` parameter that we'd like to support being resolved against the widget's state without breaking the API (without changing the parameter's type to `MaterialStateProperty<Color?>`).

This case is on the borderline, since it's a new parameter, however the existing `SnackBarThemeData.backgroundColor` is already defined as a `Color?`. So in this case, the "right" thing to do is to make this new widget parameter a `Color?` and support `MaterialStateColor` for both the theme and the widget properties. `MaterialStateColor` is-a `Color` and a `MaterialStateProperty<Color>` and can be resolved with `MaterialStateProperty.resolveAs<Color?>` :

```dart
 backgroundColor: MaterialStateProperty.resolveAs<Color?>(widget.backgroundColor)
   ?? snackBarTheme.actionBackgroundColor.resolveAs<Color?>(snackBarTheme.actionBackgroundColor)
```

The documentation for both properties should be updated to indicate that `MaterialStateColor` is supported. See for example https://api.flutter.dev/flutter/material/AppBar/backgroundColor.html.

Reply:
So I guess first a PR adding a `Color?` should be submitted and after that is merged this PR should serve as the `MaterialStateColor` support?

Reply:
This PR can add the support for a MaterialStateColor backgroundColor color per my previous comment. Another PR isn't needed.